freqtrade/freqtradebot.py

```diff
@@ -467,6 +461,28 @@ def _check_depth_of_market_buy(self, pair: str, conf: Dict) -> bool:
             logger.info(f"Bids to asks delta for {pair} does not satisfy condition.")
             return False
 
+    def _validate_stake_amount(self, pair, stake_amount, buy_limit_requested):
+        if not stake_amount:
+            logger.debug(f"Stake amount is {stake_amount}, ignoring possible trade for {pair}.")
+            return 0
+
+        min_stake_amount = self.exchange.get_min_pair_stake_amount(pair, buy_limit_requested,
+                                                                   self.strategy.stoploss)
+        max_stake_amount = self.wallets.get_available_stake_amount()
+        if min_stake_amount is not None and stake_amount < min_stake_amount:
+            stake_amount = min_stake_amount
+            logger.debug(
+                f"Stake amount for pair {pair} is too small ({stake_amount} < {min_stake_amount}), "
+                f"adjusting to {min_stake_amount}"
+            )
+        if stake_amount > max_stake_amount:
+            stake_amount = max_stake_amount
+            logger.debug(
+                f"Stake amount for pair {pair} is too big ({stake_amount} > {max_stake_amount}), "
+                f"adjusting to {max_stake_amount}"
+            )
```

Comment:
I would think that the log-level for this should be > debug (most likely info is the right value?).

Debug will only show when running freqtrade with `-v` - which will be VERY verbose as that includes a lot of other debug messages too.